packages/sol/src/verification.ts

```diff
@@ -0,0 +1,408 @@
+  password: string
+):
+  | FieldError.PasswordTooShort
+  | FieldError.PasswordTooLong
+  | FieldError.PasswordMissingLetters
+  | FieldError.PasswordMissingNumbers
+  | true {
+  if (!NUMBERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingNumbers;
+  if (!LETTERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingLetters;
+  if (password.length > MAX_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (password.length < MIN_PASSWORD_LENGTH) return FieldError.PasswordTooShort;
+  if (PASSWORD_REGEX.test(password)) return true;
+  return true;
+}
+
+export function isValidUserName(
+  userName: string
+): FieldError.InvalidUserName | true {
+  if (!USER_NAME_REGEX.test(userName)) return FieldError.InvalidUserName;
+  return true;
+}
+
+export function isValidUserBirthYear(
+  userBirthYear: number
+): FieldError.UserTooOld | FieldError.UserTooYoung | true {
+  const currentYear = new Date().getFullYear();
+  const minYear = currentYear - MAX_USER_AGE;
+  const maxYear = currentYear - MIN_USER_AGE;
+
+  if (userBirthYear < minYear) return FieldError.UserTooOld;
+  if (userBirthYear > maxYear) return FieldError.UserTooYoung;
+  return true;
+}
+
+export function isValidTutorBio(
+  bio: string
+): FieldError.EmptyBio | FieldError.TooLongBio | true {
+  if (!bio.length) return FieldError.EmptyBio;
+  if (bio.length > MAX_TUTOR_BIO_TEXT_LENGTH) return FieldError.TooLongBio;
+
+  return true;
+}
+
+export function isValidTutorAbout(
+  about: string
+):
+  | FieldError.InvalidTutorAbout
+  | FieldError.EmptyTutorAbout
+  | FieldError.TooLongTutorAbout
+  | true {
+  const tutorAboutText = getSafeInnerHtmlText(about);
+
+  if (!tutorAboutText.length) return FieldError.EmptyTutorAbout;
+  if (!HTML_REGEX.test(about)) return FieldError.InvalidTutorAbout;
+  if (tutorAboutText.length > MAX_TUTOR_ABOUT_TEXT_LENGTH)
+    return FieldError.TooLongTutorAbout;
+  return true;
+}
+
+export function isValidTutorNotice(
+  tutorNotice: number
+): FieldError.MaxNoticeExceeded | FieldError.InvalidNotice | true {
+  if (tutorNotice <= MIN_TUTOR_NOTICE_DURATION) return FieldError.InvalidNotice;
+  if (tutorNotice > MAX_TUTOR_NOTICE_DURATION)
+    return FieldError.MaxNoticeExceeded;
+  return true;
+}
+
+export function isValidRuleTitle(
+  title: string
+): FieldError.TooShortRuleTitle | FieldError.TooLongRuleTitle | true {
+  if (title.length > MAX_RULE_TITLE_LENGTH) return FieldError.TooLongRuleTitle;
+  if (title.length < MIN_RULE_TITLE_LENGTH) return FieldError.TooShortRuleTitle;
+  return true;
+}
+
+export function isValidRuleStart(
+  ruleStart: string,
+  ruleEnd: string
+):
+  | FieldError.ISOInvalidRuleStart
+  | FieldError.ISOInvalidRuleEnd
+  | FieldError.RuleStartAfterEnd
+  | FieldError.RuleStartDatePassed
+  | true {
+  if (!Date.parse(ruleStart)) return FieldError.ISOInvalidRuleStart;
+  if (!Date.parse(ruleEnd)) return FieldError.ISOInvalidRuleEnd;
+  if (new Date(ruleStart) > new Date(ruleEnd))
```

Comment:
Note that rule start and end should be plain dates and should not include time (hours, minutes, seconds). 

For example, consider these to inputs to your function `isValidRuleStart("2024-11-3 10:00", "2024-11-3 11:00")`

Your function will indicate that these are valid rule times but in reality they are the same **date**. You should only compare the dates not the datetime.  

Add regex to validate the input date format `YYYY-MM-DD`.